Source/Plugins/Reconstruction/ReconstructionFilters/PartitionGeometry.h

```diff
@@ -388,11 +402,30 @@ class Reconstruction_EXPORT PartitionGeometry : public AbstractFilter
   QString getInputHexahedralGeometryInformation() const;
   QString getInputUnknownGeometryInformation() const;
 
-  template <typename T>
-  PartitioningImageGeomResult createPartitioningSchemeGeometry(const T& geometry);
+  template <typename G>
+  void createPartitioningSchemeGeometry(const G& geometry);
+
+  void partitionCellBasedGeometry(const IGeometryGrid& geometry, Int32ArrayType& partitionIds, const std::optional<int>& outOfBoundsValue);
+  void partitionNodeBasedGeometry(const QString& geomName, const SharedVertexList& vertexList, Int32ArrayType& partitionIds, const std::optional<int>& outOfBoundsValue);
+
+  template <typename G>
+  void dataCheckPartitioningMode();
+
+  template <typename G>
+  void dataCheckBasicMode();
+
+  template <typename G>
+  void dataCheckAdvancedMode();
+
+  template <typename G>
+  void dataCheckBoundingBoxMode();
+
+  void dataCheckExistingGeometryMode();
+
+  void dataCheckNumberOfPartitions();
 
-  void partitionCellBasedGeometry(const IGeometryGrid& geometry, Int32ArrayType& partitionIds, const std::optional<int> &outOfBoundsValue);
-  void partitionNodeBasedGeometry(const QString &geomName, const SharedVertexList& vertexList, Int32ArrayType& partitionIds, const std::optional<int> &outOfBoundsValue);
+  template <typename G>
```

Comment:
Stylisticly, I would like to see ```template <typename GeomType>``` which gives the developer some indication for what type of object should be passed in.